public/controller/data_functions.php

```diff
@@ -135,44 +144,134 @@ function upload_homework($username, $semester, $course, $assignment_id, $homewor
     if (!is_open_assignment($class_config, $assignment_id)) {
         return array("error"=>"", "message"=>$assignment_id." is not a valid assignment");
     }
+    if ($num_parts !== get_num_parts($class_config, $assignment_id)) {
+        return array("error"=>"", "message"=>"Number of parts for this homework does not match configuration.  ".$num_parts." != ".get_num_parts($class_config, $assignment_id));
+    }
     $assignment_config = get_assignment_config($semester, $course, $assignment_id);
     if (!can_edit_assignment($username, $semester, $course, $assignment_id, $assignment_config)) {//Made sure the user can upload to this homework
         return array("error"=>"assignment_closed", "message"=>$assignment_id." is closed.");
     }
     //VALIDATE HOMEWORK CAN BE UPLOADED HERE
     //ex: homework number, due date, late days
 
-    $max_size = 50;
-    if (isset($assignment_config["max_submission_size"])) {
-        $max_size = $assignment_config["max_submission_size"];
-    }
+    // check if files from previous submission exists
+    if($svn_checkout == false) {
+      if(isset($previous_files)) {
 
+        // check if folder for this assignment exists
+        $assignment_path = $path_front_course."/submissions/".$assignment_id;
+        if(!file_exists($assignment_path)) {
+          display_error("Folder ".$assignment_path." from previous submission does not exist.");
```

Comment:
Yeah. The main thing is that we don't want to display path info to the user in case someone is running a non-default setup for security reason, the end user does not need to know this. I would say make sure the messages are different or at least we should consider including a unique error code if it's appropriate for them to have the same message (so we can easily find and debug it later)